openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -462,6 +468,25 @@ def post(self, request):
 
         return Response(status=status.HTTP_400_BAD_REQUEST, data=serializer.errors)
 
+    def confirm(self, request, username):
+        """
+        POST /api/user/v1/account/name_change/{username}/confirm
+
+        Confirm a name change request for the specified user, and update their profile name.
+        """
+        if not request.user.is_staff:
```

Comment:
if another part of the system invokes this, do they show up as user.is_staff or would it be something else?

Reply:
That's a good question, I'm not sure. Although I don't know if there's use case for this endpoint to be accessed by the system rather than a human - [confirm_name_change](https://github.com/openedx/edx-platform/blob/db32ff2cdf678fa8edd12c9da76a76eef0478614/common/djangoapps/student/models_api.py#L149) should probably be invoked directly for cases like that.

Reply:
I'm asking because I'm in the middle of writing down how insights flow works and insights calls the data API as itself rather than the original user.

It'll be easy to fix if we need a different check and will keep out the riffraff until then.

Reply:
grepping around I don't find request.user.is_system, mostly is_staff and a little is_superuser

Reply:
I am OK keeping this for staff only for now.